### PR TITLE
DAOS-623 build: Fix the parser of custom build targets

### PR DESCRIPTION
**Pull request by jolivier23**
Previously, the logic in the function was messed up when multiple
custom targets are used.

For example,
scons test client server

would only build the client.

This fixes the logic in the function so it handles multiple targets
appropriately.

Signed-off-by: Jeff Olivier <jeffrey.v.olivier@intel.com>

### CODE REVIEW

**Comment by jolivier23** [comment]
Test failure is DAOS-7888

File: site_scons/prereq_tools/base.py [hunk 1]
```diff
@@ -710,18 +710,20 @@ def __init__(self, env, variables, config_file=None, arch=None):
...
         self._build_targets = []
-        BUILD_TARGETS.append(build_dir)
-        if 'client' in BUILD_TARGETS:
-            self._build_targets.extend(['client'])
-        elif 'server' in BUILD_TARGETS:
-            self._build_targets.extend(['server'])
-        else:
-            # either test or default
+        check = any(item in BUILD_TARGETS for item in targets)
+        if not check or 'test' in BUILD_TARGETS:
```
Comment (ashleypittman) [inline site_scons/prereq_tools/base.py:719]: So test implies 'client' and 'server'?

File: utils/run_in_ga.sh [hunk 2]
```diff
@@ -34,7 +34,7 @@ utils/check.sh -n /opt/daos/bin/dmg
 echo ::endgroup::
 
 echo ::group::Test incremental debug build with test target
-$SCONS --jobs 10 test
+$SCONS --jobs 10 test client
```
Comment (ashleypittman) [inline utils/run_in_ga.sh:37]: if test implies client then why add it here?

File: site_scons/prereq_tools/base.py [hunk 3]
```diff
@@ -710,18 +710,20 @@ def __init__(self, env, variables, config_file=None, arch=None):
...
             self._build_targets.extend(['client', 'server', 'test'])
+        else:
+            if 'client' in BUILD_TARGETS:
+                self._build_targets.extend(['client'])
```
Comment (ashleypittman) [inline site_scons/prereq_tools/base.py:723]: There's an easier way of adding one entry, extend will join two lists.
```suggestion
                self._build_targets.append('client')
```

File: site_scons/prereq_tools/base.py [hunk 4]
```diff
@@ -710,18 +710,20 @@ def __init__(self, env, variables, config_file=None, arch=None):
...
+            if 'server' in BUILD_TARGETS:
+                self._build_targets.extend(['server'])
```
Comment (ashleypittman) [inline site_scons/prereq_tools/base.py:725]:
```suggestion
                self._build_targets.append('server')
```

File: utils/run_in_ga.sh (hunk 2 above)
Comment (jolivier23) [reply utils/run_in_ga.sh:37]: It tests that scons test client actually builds test.   With the bug, it wouldn't have done so.

File: site_scons/prereq_tools/base.py (hunk 1 above)
Comment (jolivier23) [reply site_scons/prereq_tools/base.py:719]: done

**Review by ravalsam — APPROVED** [review]

**Review by ashleypittman — APPROVED** [review]

**Pull request merged**